lib/widgets/path_editor/cards/graph_settings_card.dart

```diff
@@ -0,0 +1,295 @@
+import 'package:flutter/material.dart';
+import 'package:pathplanner/widgets/draggable_card.dart';
+import 'package:pathplanner/widgets/path_editor/editors/graph_editor.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class GraphSettingsCard extends StatefulWidget {
+  final bool holonomicMode;
+  final VoidCallback onUpdate;
+  final VoidCallback onToggleSampled;
+  final VoidCallback onToggleMinimized;
+  final GlobalKey stackKey;
+  final SharedPreferences prefs;
+  final bool isSampled;
+  final bool cardMinimized;
+
+  const GraphSettingsCard(
+      {required this.stackKey,
+      required this.holonomicMode,
+      required this.onUpdate,
+      required this.onToggleSampled,
+      required this.onToggleMinimized,
+      required this.prefs,
+      required this.isSampled,
+      required this.cardMinimized,
+      super.key});
+
+  @override
+  State<GraphSettingsCard> createState() => _GraphSettingsCardState();
+}
+
+class _GraphSettingsCardState extends State<GraphSettingsCard> {
```

Comment:
You should use booleans to represent the value of the checkboxes here. This way the app only has to rebuild the changed checkbox instead of the entire card with the UniqueKey stuff you're doing in graph_editor. This is done in generator_settings_card to make sure that the path runtime gets updated, but its not needed here. It also prevents crashes from sending giving a null value to the checkboxes due to the default values.

```suggestion
class _GraphSettingsCardState extends State<GraphSettingsCard> {
  late bool _showVelocity;
  late bool _showAccel;
  late bool _showHeading;
  late bool _showAngularVelocity;
  late bool _showCurvature;
  late bool _showRotation;
  
  @override
  void initState() {
    super.initState();
    
    _showVelocity = widget.prefs.getBool(GraphEditor.prefShowVelocity) ?? true;
    _showAcce = widget.prefs.getBool(GraphEditor.prefShowAccel) ?? true;
    _showHeading = widget.prefs.getBool(GraphEditor.prefShowheading) ?? true;
    _showAngularVelocity = widget.prefs.getBool(GraphEditor.prefShowAngularVelocity) ?? true;
    _showCurvature = widget.prefs.getBool(GraphEditor.prefShowCurvature) ?? true;
    _showRotation = widget.prefs.getBool(GraphEditor.prefShowRotation) ?? true;
  }
  
```